packages/datadog-plugin-aws-sdk/src/services/s3.js

```diff
@@ -18,6 +25,33 @@ class S3 extends BaseAwsSdkPlugin {
       bucketname: params.Bucket
     })
   }
+
+  addSpanPointer (span, response) {
+    const request = response?.request
+    const operationName = request?.operation
+    if (!['putObject', 'copyObject', 'completeMultipartUpload'].includes(operationName)) {
+      // We don't create span links for other S3 operations.
+      return
+    }
+
+    const bucketName = request?.params?.Bucket
+    const objectKey = request?.params?.Key
+    // AWS v2 (all 3 operations): `response.data.ETag`
+    // AWS v3 (putObject & completeMultipartUpload): `response.ETag`
+    // AWS v3 (copyObject): `response.CopyObjectResult.ETag`
+    const eTag = response?.data?.ETag || response?.ETag || response?.CopyObjectResult?.ETag
+
+    const pointerHash = generateS3PointerHash(bucketName, objectKey, eTag)
+    if (pointerHash) {
+      const attributes = {
+        'ptr.kind': S3_PTR_KIND,
+        'ptr.dir': SPAN_POINTER_DIRECTION.DOWNSTREAM,
+        'ptr.hash': pointerHash,
+        'link.kind': SPAN_LINK_KIND
+      }
+      span.addLink(ZERO_CONTEXT, attributes)
```

Comment:
i guess this is fine for now, but it might be nice to have an `addSpanPointer` method on the span that we can use here and in other situations? or do you want to keep that for a future refactor when we add the dynamo pointers?

Reply:
I agree. Something simple like this:
```
  addSpanPointer (attributes) {
    addLink(ZERO_CONTEXT, attributes)
  }
```

that way, we don't have to export ZERO_CONTEXT to the lambda library.